src/main/scala/org/broadinstitute/dsde/agora/server/webservice/PerRequest.scala

```diff
@@ -62,34 +66,33 @@ trait PerRequest extends Actor with LazyLogging {
 
   override val supervisorStrategy =
     OneForOneStrategy(loggingEnabled = AgoraConfig.supervisorLogging) {
+      case e => {
+        handleException(e)
+        Stop
+      }
+    }
+
+  def handleException(t: Throwable) = t match {
       //Should make a single Authorization Exception trait to minimize code duplication.
       case e: AgoraEntityAuthorizationException =>
         r.complete(Forbidden, AgoraException(e.getMessage, e.getCause, Forbidden))
-        Stop
```

Comment:
handleException is called from more than just the supervisorStrategy and in that case stop is not called anymore